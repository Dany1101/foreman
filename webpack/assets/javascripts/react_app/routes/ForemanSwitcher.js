import React from 'react';
import { Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { selectComponentByWeight } from '../components/common/Slot/SlotSelectors';

export const ForemanSwitcher = ({ children, externalRoutes }) => {
  if (Object.keys(externalRoutes).length) {
    return <Switch>{externalRoutes.concat(children)}</Switch>;
  }
  return <Switch>{children}</Switch>;
};

// map state to props
const mapStateToProps = state => ({
  externalRoutes: selectComponentByWeight('routes'),
});

// export connected component
export default connect(mapStateToProps)(ForemanSwitcher);

ForemanSwitcher.propTypes = {
  children: PropTypes.arrayOf(PropTypes.node).isRequired,
  externalRoutes: PropTypes.arrayOf(PropTypes.node),
};

ForemanSwitcher.defaultProps = {
  externalRoutes: [],
};
