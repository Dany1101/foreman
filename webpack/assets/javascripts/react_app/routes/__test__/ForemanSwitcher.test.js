import React from 'react';
import { Route } from 'react-router-dom';
import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { ForemanSwitcher } from '../ForemanSwitcher';

const children = {
  children: [<Route key="1" path="/path1" />, <Route key="2" path="path2" />],
};

const externalRoutes = [<Route key="3" path="/externalPath" />];

const fixtures = {
  'renders ForemanSwitcher with no external routes': {
    ...children,
  },
  'renders ForemanSwitcher with external routes': {
    ...children,
    externalRoutes,
  },
};
describe('ForemanSwitcher', () => {
  describe('rendering', () => {
    testComponentSnapshotsWithFixtures(ForemanSwitcher, fixtures);
  });
});
