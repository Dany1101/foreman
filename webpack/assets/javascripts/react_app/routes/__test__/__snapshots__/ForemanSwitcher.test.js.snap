// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForemanSwitcher rendering renders ForemanSwitcher with external routes 1`] = `
<Switch>
  <Route
    key="3"
    path="/externalPath"
  />
  <Route
    key="1"
    path="/path1"
  />
  <Route
    key="2"
    path="path2"
  />
</Switch>
`;

exports[`ForemanSwitcher rendering renders ForemanSwitcher with no external routes 1`] = `
<Switch>
  <Route
    key="1"
    path="/path1"
  />
  <Route
    key="2"
    path="path2"
  />
</Switch>
`;
